src/Components/bookHub/BookHub.js

```diff
@@ -73,35 +81,104 @@ function CardsDisplay() {
 
   const searchByTerm = async () => {
     let matched = [];
-    setSearchTerms(searchTerms.toLowerCase());
-    if (defaultSearch) {
+    if (category === 'title' || category === 'description' || category === 'identity') {
       // title/description/identity
       matched = allBooks.filter(isMatchTitleDescIdentity);
     } else {
-      // we can figure out how to filter both fields at once later
-      let res = await Filter('Creator', 'authored');
-      matched.push(...res);
-      res = await Filter('Creator', 'illustrated');
-      matched.push(...res);
+      let res;
+      if (category === 'author') {
+        res = await SearchFilter('Creator', 'authored');
+        matched.push(...res);
+      }
+      if (category === 'illustrator') {
+        res = await SearchFilter('Creator', 'illustrated');
+        matched.push(...res);
+      }
 
       matched = [...new Set(matched)];
     }
 
     setFilteredBooks(matched);
   };
 
+  /* We will separate useEffect for *hopefully* clearer code */
+  // Grab books from Airtable if nothing is in local state
+  useEffect(() => { if (!allBooks.length) { getCards(); } }, [allBooks]);
+
+  // Search function
   useEffect(() => {
-    if (!allBooks.length) { getCards(); }
-    if (searchTerms) {
-      (async () => searchByTerm())();
-    } else {
-      setFilteredBooks(allBooks);
+    if (alignment === 'Search') {
+      if (searchTerms) {
+        (async () => searchByTerm())();
+      } else {
+        setFilteredBooks(allBooks);
+      }
     }
-  }, [allBooks, searchTerms, defaultSearch]);
+  }, [allBooks, searchTerms, category, alignment]);
+
+  // Filter function
+  useEffect(() => {
+    if (alignment === 'Filter') {
+      const validGradeTags = gradeRangeMetadata.slice(
```

Comment:
I think that with the new airtable representation it would be faster to use comparison operators and indexes of the metadata rather than taking slices of the array and iterating over them

Reply:
Range filters arent configured in metadata